### PR TITLE
[WIP] Florian Eisenbarth pandas questions done

### PR DESCRIPTION
**Pull request by FlorianEisenbarth**

### CODE REVIEW

**Comment by tomMoral** [comment]
If you are finished, you can sign up for this RAMP challenge: https://ramp.studio/events/brevet_datacamp2020
This is what we are going to do this afternoon. The goal is to try and solve the challenge.
You need to click on the blue button: `Join the event` and I need to approve your participation so send me a message once you have done this. If you don't see the join button, please click here: https://ramp.studio/events/brevet_datacamp2020/sign_up
You can then read the starting kit and go to the left menu to submit your solutions in the sandbox.

**Comment by FlorianEisenbarth** [comment]
I send you a message to notify you that I have joined the challenge

Le mar. 10 nov. 2020 à 11:03, Thomas Moreau <notifications@github.com> a
écrit :

File: pandas_questions.py [hunk 1]
```diff
@@ -38,8 +47,26 @@ def merge_referendum_and_areas(referendum, regions_and_departments):
...
+
+    regions_and_departments.rename(
+        columns={"code_dep": "Department code"}, inplace=True)
+    regions_and_departments["Department code"] = regions_and_departments.apply(
+        func, axis=1)
+    referendum.drop(
+        referendum[
+            ((referendum["Department code"].isin(["DOM", "COM", "TOM"]))
+             | (referendum["Department code"].str.startswith("Z")))
+             ].index, inplace=True)
```
Comment (tomMoral) [inline pandas_questions.py:64]: A bit complicated to do a boolean drop the index.
You can simply select with reversed masked (use `~` for the not).

**Review by tomMoral — CHANGES_REQUESTED** [review]
Come comments to change.

File: pandas_questions.py [hunk 2]
```diff
@@ -81,3 +118,7 @@ def plot_referendum_map(referendum_result_by_regions):
 
     plot_referendum_map(referendum_results)
     plt.show()
+
+# %%
```
[the remark this answers is not in the document]
Comment (tomMoral) [reply pandas_questions.py:119]: You should remove this line to but let's say it is okay :)

File: pandas_questions.py [hunk 3]
```diff
@@ -57,12 +57,9 @@ def func(x):
...
         func, axis=1)
-    referendum.drop(
-        referendum[
-            ((referendum["Department code"].isin(["DOM", "COM", "TOM"]))
-             | (referendum["Department code"].str.startswith("Z")))
-             ].index, inplace=True)
-
+    referendum[
+        ~((referendum["Department code"].isin(["DOM", "COM", "TOM"])) |
+            (referendum["Department code"].str.startswith("Z")))]
```
Comment (tomMoral) [inline pandas_questions.py:62]: Better!

Comment (tomMoral) [reply pandas_questions.py:62]: note that this filtering is not really mandatory if you use an `inner` merge but it is nice to do it by hand!

**Review by tomMoral — APPROVED** [review]
LGTM!

**Pull request closed without merging**